src/store/inMemoryStorageService.ts

```diff
@@ -0,0 +1,32 @@
+import { StorageInterface } from '../types';
+
+// InMemoryStorageService: A service to store data in JavaScript memory.
+// It's a simple key-value storage using a Map. The data is stored as long as the page is not reloaded.
+export class InMemoryStorageService implements StorageInterface {
```

Comment:
Let's convert this to functional at some point.

Reply:
Do we see this as a frontend-only use case? If so, it might be best to expose this as a context

Reply:
I see this evolving towards sharing in context as well. We might want build a light version of react-query for actions.

Reply:
> I see this evolving towards sharing in context as well. 

React Context are a slippery slop, where folks think everything should be a context. Library should have like 90% of their state being just variable shared between functions. And for very very small exential stuff you can have Context that optimizes the Hooks story.

So before we build any React Context, it's important to ask the question, on where is the benefit for the end application. If it's just a nice to have for us as developer, than we don't need that. But again, for Hooks Rendering optimization, that we can talk about it.